presto-main/src/main/java/io/prestosql/operator/DynamicFilterCollectOperator.java

```diff
@@ -0,0 +1,191 @@
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.prestosql.operator;
+
+import com.google.common.collect.ImmutableMap;
+import io.prestosql.spi.Page;
+import io.prestosql.spi.block.Block;
+import io.prestosql.spi.connector.ColumnHandle;
+import io.prestosql.spi.predicate.Domain;
+import io.prestosql.spi.predicate.TupleDomain;
+import io.prestosql.spi.predicate.ValueSet;
+import io.prestosql.spi.type.TypeUtils;
+import io.prestosql.sql.planner.DynamicFilter;
+import io.prestosql.sql.planner.plan.PlanNodeId;
+
+import java.util.HashSet;
+import java.util.List;
+import java.util.function.Consumer;
+
+import static com.google.common.base.Verify.verify;
+import static io.prestosql.SystemSessionProperties.getDynamicFilteringMaxPerDriverByteSize;
+import static io.prestosql.SystemSessionProperties.getDynamicFilteringMaxPerDriverValueCount;
+import static java.util.Objects.requireNonNull;
+
+/**
+ * This operator acts as a simple "pass-through" pipe, while saving its input pages.
+ * The collected pages' value are used for creating a run-time filtering constraint (for probe-side table scan in an inner join).
+ * We support only small build-side pages (which should be the case when using "broadcast" join).
+ */
+public class DynamicFilterCollectOperator
+        implements Operator
+{
+    public static class DynamicFilterCollectOperatorFactory
+            implements OperatorFactory
+    {
+        private final int operatorId;
+        private final PlanNodeId planNodeId;
+        private final DynamicFilter dynamicFilter;
+
+        public DynamicFilterCollectOperatorFactory(int operatorId, PlanNodeId planNodeId, DynamicFilter dynamicFilter)
+        {
+            this.operatorId = operatorId;
+            this.planNodeId = requireNonNull(planNodeId, "planNodeId is null");
+            this.dynamicFilter = requireNonNull(dynamicFilter);
+        }
+
+        @Override
+        public Operator createOperator(DriverContext driverContext)
+        {
+            return new DynamicFilterCollectOperator(
+                    driverContext.addOperatorContext(operatorId, planNodeId, DynamicFilterCollectOperator.class.getSimpleName()),
+                    dynamicFilter);
+        }
+
+        @Override
+        public void noMoreOperators()
+        {
+        }
+
+        @Override
+        public OperatorFactory duplicate()
+        {
+            throw new UnsupportedOperationException("duplicate() is not supported for DynamicFilterCollectOperatorFactory");
+        }
+    }
+
+    private final OperatorContext context;
+    private boolean finished;
+    private Page current;
+    private final Consumer<TupleDomain<ColumnHandle>> dynamicPredicateConsumer;
+    private final int maxValueCount;
+    private final long maxByteSize;
+
+    private final List<DynamicFilter.Descriptor> descriptors;
```

Comment:
please operate on `io.prestosql.sql.DynamicFilters.Descriptor` instead since it's technically not required to filter at `ColumnHandle` level. This way we could reuse the operator for collecting dynamic filters for partition filtering in coordinator

Reply:
I switched to use `TupleDomain<Symbol>` almost everywhere, except at `LocalExecutionPlanner#LocalExecutionContext#getDynamicFilter()`.